src/main/java/io/tiledb/spark/TileDBDataReaderPartitionScan.java

```diff
@@ -476,798 +688,234 @@ private void allocateQuerybuffers(long readBufferSize) throws TileDBError {
+      // method.
+      valueVector.allocateNew();
+      valueVectorValidity.allocateNew();
+
+      createAndSetArrowBuffers(valueVector, valueVectorValidity, typeInfo, name);
     }
-    return bufferLength;
+    metricsUpdater.finish(queryAllocBufferTimerName);
   }
 
   /**
-   * This method is responsible for reading the values from the given attribute. It is only called
-   * for attributes with variable size.
+   * Creates and sets the arrowBuffers to the query.
    *
-   * @param name name of the attribute.
-   * @param dataType datatype of the attribute.
-   * @param index index of the result vector that corresponds to this attribute.
-   * @param isVar true if the attribute has variable length.
-   * @return number of values inserted.
+   * @param valueVector The main valueVector
+   * @param valueVectorValidity the helper valueVector for the validity map.
+   * @param typeInfo the typeInfo
+   * @param name current field name
    * @throws TileDBError
    */
-  private int getVarLengthAttributeColumn(
-      String name, Datatype dataType, boolean isVar, long cellValNum, int index)
+  private void createAndSetArrowBuffers(
+      ValueVector valueVector, ValueVector valueVectorValidity, TypeInfo typeInfo, String name)
       throws TileDBError {
-    metricsUpdater.startTimer(queryGetVariableLengthAttributeTimerName);
-    boolean nullable = false;
-    int numValues;
-    int bufferLength = 0;
-    short[] validityByteMap;
-    // reset columnar batch start index
-
-    if (this.arraySchema.hasAttribute(name)) {
-      try (Attribute attr = this.arraySchema.getAttribute(name)) {
-        nullable = attr.getNullable();
+    ByteBuffer data;
+    if (valueVector instanceof ListVector) {
+      ListVector lv = (ListVector) valueVector;
+      ArrowBuf arrowData = lv.getDataVector().getDataBuffer();
+      data = arrowData.nioBuffer(0, arrowData.capacity());
+      // For null list entries, TileDB-Spark will set the outer bitmap.
+      // The inner (values) bitmap should be initialized to all non-null.
+      ArrowBuf valueBitmap = lv.getDataVector().getValidityBuffer();
+      int nbytes = valueBitmap.capacity();
+      for (int i = 0; i < nbytes; i++) {
+        valueBitmap.setByte(i, 0xff);
       }
+    } else {
+      ArrowBuf arrowData = valueVector.getDataBuffer();
+      data = arrowData.nioBuffer(0, arrowData.capacity());
     }
 
-    resultVectors[index].reset();
-    resultVectors[index].getChild(0).reset();
-    bufferLength = putValuesOfAttVar(name, dataType, index);
-    if (nullable) {
-      validityByteMap = query.getValidityByteMap(name);
-      for (int i = 0; i < validityByteMap.length; i++) { // check for null values.
-        if (validityByteMap[i] == 0) resultVectors[index].putNull(i);
-      }
+    data.order(ByteOrder.LITTLE_ENDIAN);
+
+    // for nulls
+    ArrowBuf arrowBufValidity;
+    arrowBufValidity = valueVector.getValidityBuffer();
+    // necessary to populate with non-null values
```

Comment:
It is necessary to populate the validity map, otherwise no results are processed. I don't know the reason but this was the problem that I had when I first started working on this if you remember.